Framework/Core/src/TableTreeHelpers.cxx

```diff
@@ -83,312 +95,259 @@ class ColumnIterator
   // with this mArray is prepared to be used in arrow::Table::Make
   void finish();
 };
-} // namespace
 
-// is used in TableToTree
-BranchIterator::BranchIterator(TTree* tree, std::shared_ptr<arrow::ChunkedArray> col, std::shared_ptr<arrow::Field> field)
-{
-  mField = field.get();
-  mBranchName = mField->name();
+template <typename T>
+struct ROOTTypeString {
+  static constexpr char const* str = "/E";
+};
 
-  mFieldType = mField->type()->id();
-  mChunks = col->chunks();
-  mNumberChuncs = mChunks.size();
+template <>
+struct ROOTTypeString<bool> {
+  static constexpr char const* str = "/O";
+};
 
-  mLeaflistString = mBranchName;
-  mElementType = mFieldType;
-  mNumberElements = 1;
-  if (mFieldType == arrow::Type::type::FIXED_SIZE_LIST) {
+template <>
+struct ROOTTypeString<uint8_t> {
+  static constexpr char const* str = "/b";
+};
 
-    // element type
-    if (mField->type()->num_fields() <= 0) {
-      LOGP(FATAL, "Field {} of type {} has no children!", mField->name(), mField->type()->ToString().c_str());
-    }
-    mElementType = mField->type()->field(0)->type()->id();
-    // number of elements
-    mNumberElements = static_cast<const arrow::FixedSizeListType*>(mField->type().get())->list_size();
-    mLeaflistString += "[" + std::to_string(mNumberElements) + "]";
-  }
+template <>
+struct ROOTTypeString<uint16_t> {
+  static constexpr char const* str = "/s";
+};
 
-  // initialize the branch
-  mStatus = initBranch(tree);
+template <>
+struct ROOTTypeString<uint32_t> {
+  static constexpr char const* str = "/i";
+};
 
-  mCounterChunk = 0;
-  mStatus &= initDataBuffer(mCounterChunk);
-}
+template <>
+struct ROOTTypeString<uint64_t> {
+  static constexpr char const* str = "/l";
+};
 
-BranchIterator::~BranchIterator()
-{
-  delete mBranchBuffer;
+template <>
+struct ROOTTypeString<int8_t> {
+  static constexpr char const* str = "/B";
+};
 
-  delete[] mVariable_o;
-}
+template <>
+struct ROOTTypeString<int16_t> {
+  static constexpr char const* str = "/S";
+};
 
-bool BranchIterator::getStatus()
-{
-  return mStatus;
-}
+template <>
+struct ROOTTypeString<int32_t> {
+  static constexpr char const* str = "/I";
+};
 
-bool BranchIterator::initBranch(TTree* tree)
+template <>
+struct ROOTTypeString<int64_t> {
+  static constexpr char const* str = "/L";
+};
+
+template <>
+struct ROOTTypeString<float> {
+  static constexpr char const* str = "/F";
+};
+
+template <>
+struct ROOTTypeString<double> {
+  static constexpr char const* str = "/D";
+};
+
+template <typename T>
+constexpr auto ROOTTypeString_t = ROOTTypeString<T>::str;
+
+template <typename T>
+class ColumnToBranch : public ColumnToBranchBase
 {
-  // try to find branch in tree
-  mBranchPtr = tree->GetBranch(mBranchName.c_str());
-  if (mBranchPtr) {
-    return true;
+ public:
+  ColumnToBranch(TTree* tree, arrow::ChunkedArray* column, arrow::Field* field, int size = 1)
+    : ColumnToBranchBase(column, field, size)
+  {
+    if constexpr (std::is_pointer_v<T>) {
```

Comment:
This can be expressed as well using field, no? Why do you need it to be constexpr?